### PR TITLE
Gradle modules refactor, spotless formatter

### PR DESCRIPTION
**Pull request by kaylendog**
StickyAPI can be considered a modular API - we have the `common`, `bukkit`, and `bungeecord` packages. Ideally, `common` should not reference methods from either `bukkit` or `bungeecord`, but vice versa, the two environment-specific packages should. This is the pefcet situation in which we should be using Gradle Multi-module Projects.

Additionally, formatting has been an issue for quite some time. I have solved this by adding [spotless](https://github.com/diffplug/spotless/tree/main/plugin-gradle) to the Gradle build process. Running the [prettier plugin](https://github.com/diffplug/spotless/tree/main/plugin-gradle#prettier) on the backend. 

Will definitely need some input on what formatting rules we want to abide by as a team. I will be adding spotless to the template projects for later use.

### CODE REVIEW

**Comment by kaylendog** [comment]
@Prouser123, needs some work before GHA is ready to artifact the project.

File: build.gradle [hunk 1]
```diff
@@ -2,73 +2,61 @@ plugins {
...
 
-    // Mocking
-    testImplementation("org.powermock:powermock-module-junit4:2.0.9")
-    testImplementation("org.powermock:powermock-api-mockito2:2.0.9")
-    testImplementation("org.powermock:powermock-module-junit4-rule:2.0.9")
-    testImplementation("org.powermock:powermock-classloading-xstream:2.0.9")
-    testImplementation("org.mockito:mockito-core:3.6.28")
+        java {
+            importOrder()
+            prettier(['prettier': '2.0.5', 'prettier-plugin-java': '0.8.0']).config(['parser': 'java', 'tabWidth': 4])
```
Comment (vladfrangu) [inline build.gradle:49]: Isn't the default printWidth like.. 60? Can we please bump it to like.. 100/120? Of course, open to discussions

Comment (ZachyFoxx) [reply build.gradle:49]: +1, would be better at 100

**Comment by kaylendog** [comment]
Closing as this is tracking too far behind master.

**Pull request closed without merging**